packages/calendar-time-utils/src/calendar-time.ts

```diff
@@ -162,3 +163,57 @@ export const parseInputText = (
   }
   return '';
 };
+
+const localizedDateFormatPatternCache = new Map<string, string>();
+export const getLocalizedDateTimeFormatPattern = (
+  locale: string,
+  formatType: 'date' | 'time' | 'full' = 'date'
+): string => {
+  const key = `${locale}::${formatType}`;
+  if (localizedDateFormatPatternCache.has(key)) {
+    return localizedDateFormatPatternCache.get(key)!;
+  }
```

Comment:
Since this function will be used from the components every time they are rendered, I thought it would help to have a simple cache here to avoid calculating the pattern too many times.

Reply:
Sure, makes sense. Thanks